### PR TITLE
Minor rsx fixes

### PR DESCRIPTION
**Pull request by kd-11**
- Workaround for src type 3 as we gather more information
- Workaround for invalid/unknown methods
- Fix TXP when performed on a cubemap texture (why is it called in the first place?)
- Do not always assume a tex variable exists when doing alphakill checks. Variable may have been optimized away.

### CODE REVIEW

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 1]
```diff
@@ -29,9 +29,11 @@ namespace rsx
 	
 	std::array<rsx_method_t, 0x10000 / 4> methods{};
 
-	[[noreturn]] void invalid_method(thread*, u32 _reg, u32 arg)
+	void invalid_method(thread*, u32 _reg, u32 arg)
 	{
-		fmt::throw_exception("Invalid RSX method 0x%x (arg=0x%x)" HERE, _reg << 2, arg);
+		//Don't throw, gather information and ignore broken/garbage commands
+		//TODO: Investigate why these commands are executed at all. (Heap corruption? Alignment padding?)
+		LOG_ERROR(RSX, "Invalid RSX method 0x%x (arg=0x%x)" HERE, _reg << 2, arg);
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: I still don't think it's a good idea. One of the possible reason is handling NOPs.

Comment (kd-11) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: Thought so as well, but the range of values we're seeing is too large. I'm convinced some are from broken command buffers.
It's difficult to say what happens if you pass broken instructions to a real PS3, I'm guessing they'll be ignored. For some games, the instructions seem to have no effect on visual output (probably undocumented NOPs). We only have one NOP documented, (0x1718 >> 2) and it is handled.

**Comment by JohnGodgames** [comment]
Test results from https://github.com/kd-11/rpcs3/commit/7ec8d496272f91393b6b505656e5cbda080d5673

**Wheelman now goes ingame**
wheelman 1

**Tales of Xillia now goes ingame**
xillia 1
xillia 2

**Comment by raven02** [comment]
Just remember one regression from Iron man 2 

```
E {rsx::thread} RSX: Failed to compile shader: 0(63) : error C0000: syntax error, unexpected reserved word "break", expecting "::" at token "break"
0(64) : error C0000: syntax error, unexpected reserved word "break", expecting "::" at token "break"
```

**Comment by kd-11** [comment]
@raven02 OGL Log?

**Comment by raven02** [comment]
@kd-11 https://mega.nz/#!DF83QKzJ!0GUEEBY09GwxCrOzNfRZvETcPqZ3Nm20jAacWyJgPl0

**Comment by RainKikyou** [comment]
Tales of Xillia 2 also normal，no longer crash

**Comment by kd-11** [comment]
@raven02 Is iron man 2 still broken? I see a screenshot claiming its fixed.

**Comment by raven02** [comment]
I see. Then probably fixed if screenshot confirmed.

**Comment by slashiee** [comment]
Sonic Generations now gets to loading

**Comment by AniLeo** [comment]
Sonic Generations already went to Intro before

**Comment by OverlordCW** [comment]
In RDR I now get this:
E {rsx::thread} RSX: Unknown/illegal instruction: 0x3d (forced unit 2)
E {rsx::thread} RSX: Invalid RSX method 0xb9f8 (arg=0x0)
E {rsx::thread} RSX: Invalid RSX method 0x6d08 (arg=0x0)
E {rsx::thread} RSX: Invalid RSX method 0x9810 (arg=0x0)

**Comment by Sanada625** [comment]
can someone help me
why in kamen rider battride war 2 my main character not show up in monitor ????

**Pull request merged**